src/index.js

```diff
@@ -3,30 +3,79 @@ var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
 var API = 'https://rickandmortyapi.com/api/character/';
 var xhttp = new XMLHttpRequest();
 
-function fetchData(url_api, callback) {
+var dataJson
+var data2Json
+
+//nueva version: promise
+const fetchData = (url_api) => {
+  
+  return new Promise((resolve, reject) => {
+
+    xhttp.onreadystatechange = function (event) {
+      if (xhttp.readyState === 4) {
+        if (xhttp.status == 200)
+          resolve(xhttp.responseText);
+        else reject(url_api);
```

Comment:
Te recomiendo ser mas claros en el reject, esto te permite identificar el error mas fácil. 

```suggestion
          reject(new Error("Algo salió mal."));
```